lib/jeff.ex

```diff
@@ -174,6 +176,79 @@ defmodule Jeff do
     ACU.send_command(acu, address, MFG, params) |> handle_reply()
   end
 
+  @doc """
+  Send file data to a PD
+  """
+  @spec file_transfer(acu(), osdp_address(), binary()) ::
+          Reply.FileTransferStatus.t() | cmd_err()
+  def file_transfer(acu, address, data) when is_binary(data) do
+    file_transfer(acu, address, data, nil)
+  end
+
+  @spec file_transfer(acu(), osdp_address(), binary(), function() | nil) ::
+          Reply.FileTransferStatus.t() | cmd_err()
+  def file_transfer(acu, address, data, progress_callback) when is_binary(data) do
+    total_packets = div(byte_size(data) + @osdp_max_packet_size - 1, @osdp_max_packet_size)
+
+    FileTransfer.command_set(data, @osdp_max_packet_size)
+    |> run_file_transfer(acu, address, progress_callback, 0, total_packets)
```

Comment:
Total nit - But you could avoid passing around the `packet_num` and instead deduce when needed

```elixir
packet_num = total_packets - length(rem)
```